src/IW/Core/WithId.hs

```diff
@@ -0,0 +1,20 @@
+{-# LANGUAGE DeriveAnyClass #-}
+
+module IW.Core.WithId 
+       ( WithId (..)
+       ) where
+
+import IW.Core.Id (Id (..))
+
+
+data WithId a = WithId
+    { withIdId  :: !(Id a)
+    , withIdVal :: !a
+    } deriving stock (Generic, Show, Eq)
+      deriving anyclass (ToJSON)
+    
+instance FromRow a => FromRow (WithId a) where
+    fromRow = WithId <$> field <*> fromRow
+
+instance ToRow a => ToRow (WithId a) where
+    toRow WithId{..} = toRow withIdVal  
```

Comment:
Here it should be:

```suggestion
    toRow WithId{..} = toRow (Only withIdId :. withIdVal)  
```

So `WithId` actually outputs `Id` to row inside SQL query. If you don't need to convert `Id`, you just extract `withIdVal` and call `toRow` for this data type.

Reply:
Yeah right that’s what we were trying avoid initially. Makes sense👍🏽